datarepo-clienttests/src/main/java/scripts/testscripts/RetrieveSnapshot.java

```diff
@@ -34,14 +42,52 @@ public RetrieveSnapshot() {
 
   private SnapshotSummaryModel snapshotSummaryModel;
   private List<BlobId> scratchFiles = new ArrayList<>();
+  private boolean deleteProfile;
 
   public void setup(List<TestUserSpecification> testUsers) throws Exception {
-    // create the profile and dataset
-    super.setup(testUsers);
+    // Specify the name of a profile to use for the test with the TEST_RUNNER_BILLING_PROFILE_NAME
+    // env variable.
+    // If it's not specified or the profile isn't found, a new one with a randomly generated name
+    // will be created
+    String profileNameRaw = System.getenv("TEST_RUNNER_BILLING_PROFILE_NAME");
+    Optional<String> profileName;
+    if (StringUtils.isBlank(profileNameRaw)) {
+      profileName = Optional.empty();
+      deleteProfile = true;
+    } else {
+      profileName = Optional.ofNullable(profileNameRaw);
+      deleteProfile = false;
+    }
+
+    // pick the a user that is a Data Repo steward to be the dataset creator
+    datasetCreator = SAMUtils.findTestUserThatIsDataRepoSteward(testUsers, server);
 
     // get the ApiClient for the snapshot creator, same as the dataset creator
     ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
     RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+    ResourcesApi resourcesApi = new ResourcesApi(datasetCreatorClient);
+    BillingProfileModel billingProfileModel =
+        profileName
```

Comment:
will `profileName` ever be more than one item? If not, is there an advantage using an option and then `map` compared to an `if (profileName != null) { try ... }`

Reply:
I think it would only ever be only item since it comes from a single string env variable and I'm not sure what it would mean for it to be more than one item